sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/AppConfigurationPropertySource.java

```diff
@@ -145,27 +151,26 @@ public Object getProperty(String name) {
      * </p>
      *
      * <p>
-     * <b>Note</b>: Doesn't update Feature Management, just stores values in cache.
-     * Call {@code initFeatures} to update Feature Management, but make sure its
-     * done in the last {@code AppConfigurationPropertySource}
+     * <b>Note</b>: Doesn't update Feature Management, just stores values in cache. Call {@code initFeatures} to update
+     * Feature Management, but make sure its done in the last {@code AppConfigurationPropertySource}
      * AppConfigurationPropertySource}
      * </p>
      *
      * @param featureSet The set of Feature Management Flags from various config stores.
      * @return Updated Feature Set from Property Source
      * @throws IOException Thrown when processing key/value failed when reading feature flags
+     * @throws AppConfigurationStatusException An error occurred in connecting, and should be retried on a replica if
+     * possible.
      */
-    FeatureSet initProperties(FeatureSet featureSet) throws IOException {
-        String storeName = configStore.getEndpoint();
+    FeatureSet initProperties(FeatureSet featureSet) throws IOException, AppConfigurationStatusException {
         SettingSelector settingSelector = new SettingSelector();
 
         PagedIterable<ConfigurationSetting> features = null;
         // Reading In Features
-        if (configStore.getFeatureFlags().getEnabled()) {
-            settingSelector.setKeyFilter(configStore.getFeatureFlags().getKeyFilter())
-                .setLabelFilter(configStore.getFeatureFlags().getLabelFilter());
-            features = clients.listSettings(settingSelector, storeName);
+        if (featureStore.getEnabled()) {
```

Comment:
Is it possible that the `enabled` will be null? Should we use `Boolean.True.equals(feature.store.getEnabled())` instead?

Reply:
It shouldn't it defaults to false, https://github.com/mrm9084/azure-sdk-for-java/blob/fd02a0f40f859c65fcf6571fdbbf4e06c480c195/sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/properties/FeatureFlagStore.java#L20